lib/vdsm/storage/exception.py

```diff
@@ -115,6 +115,22 @@ def __init__(self, volume, *args):
             self.value += list(args)
 
 
+class VolumeLVMCommandError(StorageException):
+    code = 4001
+    msg = "Volume exception with LVM error"
+
+    def __init__(self, error=None):
+        if error is not None and not isinstance(error, LVMCommandError):
+            raise TypeError(
+                f"Expecting instance of LVMCommandError, got {type(error)}")
+
+        self.error = error
```

Comment:
Because we use the dict order, the order of the parameter depends now on the way
the sub class is implemented. For example:

    class ErrorIsFirst(VolumeLVMCommandError):
    
        def __init__(self, name, error=None):
            super().__init__(error)
            self.name = name

    class ErrorIsLast(VolumeLVMCommandError):
    
        def __init__(self, name, error=None):
            self.name = name
            super().__init__(error)

This is not that bad, but kind of surprising. I think it is better if we
control the formatting so it is not effected by the way the class is initialized.